lua/dylib.lua

```diff
@@ -30,6 +68,55 @@ int add_fib_entry(const char *prefix, const char *gt_ip, const char *gw_ip,
 	enum gk_fib_action action, struct gk_config *gk_conf);
 int del_fib_entry(const char *ip_prefix, struct gk_config *gk_conf);
 
+size_t strlen(const char *s);
+
 ]]
 
 c = ffi.C
+
+-- The following is an example function that can be used as
+-- the callback function of list_fib4_entries() and list_fib6_entries().
+
+-- Parameter fib_dump_entry is going to be released after
+-- print_fib_dump_entry() returns, so don't keep references to fib_dump_entry
+-- or any of the data reachable through its fields.
+
+function print_fib_dump_entry(fib_dump_entry)
+	local i = 0
+	local res = ""
+	local entry = ffi.cast("struct gk_fib_dump_entry *", fib_dump_entry)
+	local ip_addr_str
+
+	ip_addr_str = dylib.ip_format_addr(entry.addr)
+	res = res .. "FIB entry for IP prefix: " .. ip_addr_str ..
+		"/" .. entry.prefix_len .. " with action " ..
+		tostring(entry.action)
+
+	if entry.action == c.GK_FWD_GRANTOR then
+		ip_addr_str = dylib.ip_format_addr(entry.grantor_ip)
+		res = res .. "\n\tGrantor IP address: " .. ip_addr_str
+	end
+
+	res = res .. "\n\t#Ethernet cache entries: " .. tostring(entry.num_ether_entries)
+
+	while (i < entry.num_ether_entries) do
+		local d_buf
+		local eth_entry = entry.ether_tbl + i
+		local stale
+		if eth_entry.stale then
+			stale = "stale"
+		else
+			state = "fresh"
+		end
+
+		ip_addr_str = dylib.ip_format_addr(eth_entry.nexthop_ip)
+		d_buf = dylib.ether_format_addr(eth_entry.d_addr)
+		res = res .. "\n\t\tEther cache entry#" .. tostring(i + 1) .. ": [state: " ..
```

Comment:
I think you don't need `tostring()` here as well. But keep the parenthesis, that is, keep `(i + 1)`.